microsoft-edge/webview2/concepts/user-data-folder.md

```diff
@@ -1,62 +1,166 @@
+-->
+
+As a WebView2 developer, you need to know about the user data folder for WebView2 apps, including how it's created, how it's used, and how you need to handle it.
+
+WebView2 apps use user data folders to store browser data, such as cookies, permissions, and cached resources.
+*  Each instance of a WebView2 control is associated with a user data folder.
+*  Each user data folder is unique to a user.
 
 
 <!-- ====================================================================== -->
-## Best practices
+## How to create user data folders
+
+By default, WebView2 creates a user data folder in the default location.  The default location works well for most apps.  If your app has specific needs, you can specify a different location.  Make sure that user data folder location has appropriate Read/Write permissions for the WebView2 app runtime.
+
+To specify the location of the user data folder, include the `userDataFolder` parameter when calling [ICoreWebView2Environment](/microsoft-edge/webview2/reference/win32/icorewebview2environment) (Win32) or [CoreWebView2Environment](/dotnet/api/microsoft.web.webview2.core.corewebview2environment) (.NET).  After creation, browser data from your WebView2 control is stored in a subfolder of `userDataFolder`.
 
-User data folders are created automatically by WebView2.  WebView2 developers control the lifetime of the user data folder.  If your application re-uses user data from application sessions, consider saving the user data folders, otherwise you can delete them.  Consider the following scenarios when deciding how to manage your user data folders:
+When `userDataFolder` isn't specified, WebView2 creates user data folders at default locations as follows:
 
-*  If the same user uses your application repeatedly, and the web content of the application relies on the user's data, save the user data folder.
+*  For packaged Windows Store apps, the default user folder is the `ApplicationData\LocalFolder` subfolder in the package's folder.
 
-*  If multiple users use your application repeatedly, create a new user data folder for each new user, and save the user data folder of each user.
+*  For existing desktop apps, the default user data folder is the executable (`exe`) path of your app + `.WebView2`.
+   *  Instead of using the default location of the user data folder, we recommend that you explicitly specify its location.
+   *  We recommend creating the user data folder in the same folder where all other app data is stored.
 
-*  If your application doesn't have repeat users, create a new user data folder for each user, and delete the previous user data folder.
+*  For WinUI 2 & WinUI3, instead of the above approaches, you use the environment variable: `WEBVIEW2_USER_DATA_FOLDER="\<folder path\>"`.  See [CoreWebView2Environment.UserDataFolder Property](/dotnet/api/microsoft.web.webview2.core.corewebview2environment.userdatafolder).
```

Comment:
```suggestion
*  For WinUI 2 & WinUI3, instead of the above approaches, you use the environment variable: `WEBVIEW2_USER_DATA_FOLDER="\<folder path\>"`. 
```

Should we remove this link to the dotnet environment interface? This is not available in winui2/3 as far as I'm aware.

Reply:
@jasonstephen15 - Here's text I'm cleaning up in the .md file here.  I plan to reveal the "winrt" URL and delete/hide at bottom, the dotnet URL:

tag PM in PR to validate the above, and the following link(s).  still self-contained (in a way , for dependencies, for deployment (dlls)) though not packaged into a single file? -->

"instead of the above approaches, you use the environment variable: `WEBVIEW2_USER_DATA_FOLDER="\<folder path\>"`.  See:

.net (bad):
[CoreWebView2Environment.UserDataFolder Property](http://docs.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2environment.userdatafolder). -->

WinRT (good):
* [CoreWebView2Environment.UserDataFolder](https://docs.microsoft.com/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2environment#userdatafolder)